libs/common/ConvertUtils.cs

```diff
@@ -53,5 +54,38 @@ public static void MakeUpperCase(Span<byte> command)
                 if (c > 96 && c < 123)
                     c -= 32;
         }
+
+        /// <summary>
+        /// Check if two byte spans are equal, where right is an all-upper-case span, ignoring case if there are ASCII bytes.
+        /// </summary>
+        /// <param name="left"></param>
+        /// <param name="right"></param>
+        /// <returns></returns>
+        public static bool EqualsUpperCaseSpanIgnoringCase(this ReadOnlySpan<byte> left, ReadOnlySpan<byte> right)
+        {
+            if (left.SequenceEqual(right))
```

Comment:
Isn't it wasteful to do a SequenceEqual here? Wouldn't it be better to just iterate over the array once the way we're doing at the bottom?

Reply:
this is optimizing for the common case where sequences are equal, as span.sequenceequal does some tricks to make this very fast.